public/js/utils/utils.js

```diff
@@ -172,6 +172,18 @@ function updateObj<T>(obj: T, fields: $Shape<T>) : T {
   return Object.assign({}, obj, fields);
 }
 
+function throttle(func: any, ms: number) {
+  let timeout;
+  return (...args: any) => {
+    if (!timeout) {
```

Comment:
I don't prefer early returns for small functions, but I know that's a stylistic choice. It's easier to see the full logic instead of having to see what returns at the top, but doing it for large/complex functions is fine... It's common in style guides to avoid early returning, but we should solidify these style rules at some point.